ui/imports/shared/popups/keycard/helpers/KeycardItem.qml

```diff
@@ -0,0 +1,115 @@
+import QtQuick 2.14
+import QtQml.Models 2.14
+import QtQuick.Controls 2.14
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Core.Utils 0.1
+import StatusQ.Components 0.1
+import StatusQ.Controls 0.1
+
+import utils 1.0
+
+StatusListItem {
+    id: root
+
+    property string keycardName: ""
+    property bool keycardLocked: false
+    property int keyPairType: Constants.keycard.keyPairType.unknown
+    property string keyPairIcon: ""
+    property string keyPairImage: ""
+    property string keyPairAccounts: ""
+
+    signal keycardSelected()
+
+    color: root.keycardLocked? Theme.palette.dangerColor3 : Style.current.grey
+    title: root.keycardName
+    statusListItemTitleAside.textFormat: Text.RichText
+    statusListItemTitleAside.visible: true
+    statusListItemTitleAside.text: {
+        let t = ""
+        if (root.keyPairType === Constants.keycard.keyPairType.profile) {
+            t = Utils.getElidedCompressedPk(d.myPublicKey)
+        }
+        if (root.keycardLocked) {
+            let label = qsTr("Keycard Locked")
+            t += `<font color="${Theme.palette.dangerColor1}" size="5">${label}</font>`
+        }
+        return t
+    }
+
+    asset {
+        width: root.keyPairIcon? 24 : 40
+        height: root.keyPairIcon? 24 : 40
+        name: root.keyPairImage? root.keyPairImage : root.keyPairIcon
+        isImage: !!root.keyPairImage
+        color: root.keyPairType === Constants.keycard.keyPairType.profile?
+                   Utils.colorForPubkey(d.myPublicKey) : Theme.palette.primaryColor1
+        letterSize: Math.max(4, asset.width / 2.4)
+        charactersLen: 2
+        isLetterIdenticon: !root.keyPairIcon && !asset.name.toString()
+        bgColor: root.keycardLocked? Theme.palette.dangerColor2 : Theme.palette.primaryColor3
+    }
+
+    ringSettings {
+        ringSpecModel: root.keyPairType === Constants.keycard.keyPairType.profile?
+                           Utils.getColorHashAsJson(d.myPublicKey) : []
```

Comment:
Should it or not display the ring with an ENS name?

Reply:
@caybro `Utils.getColorHashAsJson` handles that.